qiskit_experiments/framework/analysis_result_data.py

```diff
@@ -0,0 +1,49 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Helper dataclass for constructing analysis results."""
+
+import dataclasses
+import logging
+from typing import Optional, Dict, Any, List
+
+LOG = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class AnalysisResultData:
+    """Dataclass for experiment analysis results"""
+
+    # TODO: move stderr and unit into custom value class
+    name: str
+    value: Any
+    chisq: Optional[float] = None
```

Comment:
This is not for this PR but it would be great if this is more general quantity such as `evaluate = {"name": "chisq", "value": 0.12345}` (or some class object rather than dict), e.g. this can be confidence for quantum volume.

Reply:
This would probably require updates to DB service since chisq is an expected field, but could be worth discussing with them.